Fixes #5388 -- Added Documentation for constraint while using Conda (#6509)
readthedocs/readthedocs.org: docs/config-file/v2.rst

:Type: ``number``
:Default: ``3``

> .. note::

Reviewer: ```suggestion
.. warning::
```

>   If you are using :ref:`conda <config-file/v2:conda>` environment to manage

Reviewer: ```suggestion
  If you are using a :ref:`Conda <config-file/v2:conda>` environment to manage
```

>   the build, this version number will not have any effect, as it is managed by conda.

Reviewer: ```suggestion
  the build, this setting will not have any effect, as the Python version is managed by Conda.
```

>   the build, this setting will not have any effect, as the Python version is managed by Conda.

Reviewer 2: @stsewd what about making these notes as validation errors when the build is triggered? I feel that it only adds confusion to users the ability to use both at the same time and do not receive any error when building?

Reviewer: Yeah, we can raise an error in addition to just put it in the docs

Author: @humitos I think your question is answered in the first line of the description of  #5388 

> We should mention the restrictions of using conda, a better approach would be to raise an exception in the validation process, but that would break some projects, so that's not possible now.

Reviewer: https://github.com/readthedocs/readthedocs.org/issues/6569

Reviewer: @Parth1811 thanks for pointing that, didn't remember I wrote that :smile:

         - requirements: docs/requirements.txt
         - requirements: requirements.txt

.. warning::

  If you are using a :ref:`Conda <config-file/v2:conda>` environment to

>   manage build, this setting will not have any effect. Instead

Reviewer: ```suggestion
  manage the build, this setting will not have any effect. Instead
```

Author: Hopefully, that's all :)

>   manage build, then this setting will not have any effect. Instead

Reviewer: ```suggestion
  manage the build, this setting will not have any effect. Instead
```

>   add the extra requirements to ``environment`` file of Conda.

Reviewer: ```suggestion
  add the extra requirements to the ``environment`` file of Conda.
```

>   add the extra requirements to `environment` file of Conda.

Reviewer: ```suggestion
  add the extra requirements to ``environment`` file of Conda.
```

         - requirements: docs/requirements.txt
         - requirements: requirements.txt

.. warning::

>   If you use a :ref:`conda <config-file/v2:conda>` environment,

Reviewer: ```suggestion
  If you are using a :ref:`Conda <config-file/v2:conda>` environment to manage the build,
```

  then requirements will not be installed from the file mentioned in
  `requirements`, You have to put the pip requirements inside the

>   `environment.yml` used by conda.

Reviewer: This can be written like

```
this setting will not have any effect. Use Conda to install extra requirements.
```

That you can access to them by enabling this option.
See :ref:`builds:The build environment` for more details.

.. warning::

  If you are using a :ref:`Conda <config-file/v2:conda>` environment
  to manage build, you should not set `system_packages` to `True`,
  as this setting will not have any effect, since the virtual

>   environment creation is managed by Conda.

Reviewer: This can be written as

```
If you are using a :ref:`Conda <config-file/v2:conda>` environment to manage the build,
this setting will not have any effect, since the virtual environment creation is managed by Conda.
```

That you can access to them by enabling this option.
See :ref:`builds:The build environment` for more details.

.. warning::

>   If you install your project with :ref:`conda <config-file/v2:conda>`,

Reviewer: ```suggestion
  If you are using a :ref:`Conda <config-file/v2:conda>` environment to manage the build,
```

  you should not set `system_packages` to `True`. Because the virtual environment

>   creation in managed by conda. So, `syste_packages` does't have any effect.

Reviewer: ```
this setting will not have any effect, since the virtual environment creation is managed by Conda.
```
